doc/aspice_swe3_4/example/iceoryx_component/include/example_component/example_derived_class.hpp

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef IOX_DOC_EXAMPLE_COMPONENT_EXAMPLE_DERIVED_CLASS_HPP
+#define IOX_DOC_EXAMPLE_COMPONENT_EXAMPLE_DERIVED_CLASS_HPP
+
+#include "source/example_base_class.hpp"
+
+
+namespace example
+{
+/// @brief Short description
+/// @code
+///   minimalistic working example which uses all public methods
+/// @endcode
+/// @swcomponent example_component
+void someFreeFunction();
+
+/// @brief Forward declaration of other class
+class SomeOtherClass;
+
+/// @brief Short description
+/// @details Detailed description
+/// @code
+///
+/// @endcode
+/// @note Important note for user/developer
+/// @swcomponent cpp
+template <typename T>
+class ExampleDerivedClass : public ExampleBaseClass<T>
+{
+  public:
+    /// @brief Short description
+    /// @details Detailed description
+    /// @param[in] a Description of input parameter a
+    /// @param[in] b Description of input parameter b
+    ExampleDerivedClass(const uint64_t a, const uint64_t b) noexcept;
+
+    /// @brief Short description
+    void simpleMethod() const noexcept;
+
+    /// @brief Short description
+    /// @param[in] c Description of input parameter c
+    /// @param[out] d Description of output parameter d
+    /// @return Description of return value
+    uint64_t complexMethod(const uint32_t c, const uint32_t* d) noexcept;
```

Comment:
Personally I see no point of having value paramters with const qualifier. Compiler optimization is an illusion in this regard and restricting the implementation to not being able to change its *local copy* is useless in my opinion. But as long as this is not mandatory I am cool with it. ;-) Just food for thought. I will not point this out elsewhere.

Reply:
i will add a reference in the next PR